Clean up examples and add puppeteer tests

blink1073 opened the pull request:
<!--
Thanks for contributing to JupyterLab!
Please fill out the following items to submit a pull request.
See the contributing guidelines for more information:
https://github.com/jupyterlab/jupyterlab/blob/master/CONTRIBUTING.md
-->

## References
Fixes #5819
<!-- Note issue numbers this pull request addresses (should be at least one, see contributing guidelines above). -->

<!-- Note any other pull requests that address this issue and how this pull request is different. -->

## Code changes
Update example  tsconfig and webpack.   Cleans up all of the examples to be browser-testable and adds test for all 10 examples.

<!-- Describe the code changes and how they address the issue. -->

## User-facing changes
All of the examples now build and run without error :smile:

<!-- Describe any visual or user interaction changes and how they address the issue. -->

<!-- For visual changes, include before and after screenshots here. -->

## Backwards-incompatible changes
NA
<!-- Describe any backwards-incompatible changes to JupyterLab public APIs. -->

ian-r-rose commented on scripts/travis_script.sh:
```diff
@@ -26,6 +26,17 @@ if [[ $GROUP == js ]]; then
+    sleep 5
+    kill $TASK_PID
+    wait $TASK_PID
+
```
Should we `popd` here? Doesn't matter much right now since it's at the end, but could in the future.

ian-r-rose commented:
I took a pass at the other services tests and ran into a couple of issues:
1. The `browser` example currently fails because `tsc` is missing from its dependencies in the `package.json`. It might pass locally if the user has a reasonably up-to-date `tsc` installed globally.
2. I couldn't get the `node` example working because of an `_xsrf` error. I'm not sure why this is happening, since the token is an empty string in this example. Possibly related to #6141?

blink1073 commented:
Thanks for the testing!  I'll dig into those other ones as well.

blink1073 commented:
@ian-r-rose, the xrsf issue arose if you ran that script from a different directory, now fixed.

I went ahead and added automated testing for all 10 examples :smile:

vidartf commented on examples/example_check.py:
```diff
@@ -0,0 +1,60 @@
+
```
Would you mind adding a small docstring explaining what the purpose of this file is? It makes it a lot easier to navigate the helper files!

vidartf commented on examples/test_examples.py:
```diff
@@ -0,0 +1,45 @@
+
```
Same here, a docstring would be helpful.

blink1073 commented:
Good call, docstrings added.

vidartf commented:
LGTM!

SimonBiggs commented on packages/services/examples/typescript-browser-with-output/src/index.ts:
```diff
@@ -16,25 +17,20 @@ import {
+  outputArea.future = kernel.requestExecute({ code });
+  document.getElementById('outputarea').appendChild(outputArea.node);
+  await outputArea.future.done;
+  console.log('Test complete!');
```
Thanks for this @blink1073 :). Really helps me jump back into programming with @jupyterlab tools with the new changes :).

Thank you!

blink1073 replied:
No problem @SimonBiggs, nice to e-see you again!

SimonBiggs replied:
:). Also, this does remind me, I do have quite a bit else to thank you for. I learnt so much off of all the work you've done here. For example I have used what I learnt about Lerna here to implement a Python monorepo https://github.com/pymedphys/pymedphys/tree/master/packages. I couldn't have done it without your example, so thank you :).

I also got the ideas of dependency graphing off of you guys:

https://pymedphys.com/developer/dependencies.html

And many, many other things. So, thank you. :) Thanks @blink1073 :).

blink1073 replied:
My pleasure.  One day we'll meet and you can buy me a beer.  ;)

SimonBiggs replied:
Now that, is a good plan :)
